centaur/src/main/scala/centaur/json/JsonUtils.scala

```diff
@@ -71,9 +91,13 @@ object JsonUtils {
     def nonEmptyObjectArray = jsArray.isObjectArray && jsArray.nonEmpty
     def isSingleCallArray = jsArray.hasShardIndex && jsArray.size == 1
 
-    def hasShardIndex: Boolean = {
-      if (jsArray.nonEmptyObjectArray) jsArray.elements.map(_.asJsObject) forall { _.hasShardIndex }
+    def hasField(f: JsObject => Boolean): Boolean = {
```

Comment:
I would make this private since it's a helper for the `hasShardIndex` and `hasAttemptNumber` methods